packages/grid/x-data-grid/src/internals/components/base/GridOverlayWrapper.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material/utils';
+import { useGridSelector } from '../../hooks/utils/useGridSelector';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { gridDensityHeaderHeightSelector } from '../../hooks/features/density/densitySelector';
+import { GridEvents } from '../../models/events';
+
+export function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
+  const apiRef = useGridApiContext();
+  const rootProps = useGridRootProps();
+  const headerHeight = useGridSelector(apiRef, gridDensityHeaderHeightSelector);
+
+  const [viewportInnerSize, setViewportInnerSize] = React.useState(
+    () => apiRef.current.getRootDimensions()?.viewportInnerSize ?? null,
+  );
+
+  const handleViewportSizeChange = React.useCallback(() => {
+    setViewportInnerSize(apiRef.current.getRootDimensions()?.viewportInnerSize ?? null);
+  }, [apiRef]);
+
+  useEnhancedEffect(() => {
+    return apiRef.current.subscribeEvent(
+      GridEvents.viewportInnerSizeChange,
+      handleViewportSizeChange,
+    );
+  }, [apiRef, handleViewportSizeChange]);
+
+  let height: React.CSSProperties['height'] = viewportInnerSize?.height ?? 0;
+  if (rootProps.autoHeight && height === 0) {
+    height = 'auto';
+  }
+
+  if (!viewportInnerSize) {
+    return null;
+  }
+
+  return (
+    <div
+      style={{
+        height,
+        width: viewportInnerSize?.width ?? 0,
+        position: 'absolute',
+        top: headerHeight,
```

Comment:
Looking at #4026, I think we should set `top: 0` and full height (including footer) for `ErrorOverlay`, since neither grid column header nor footer are rendered.
At this point, I don't think we need `GridOverlayWrapper` for `ErrorOverlay` at all, since scrolling is not an issue if there's an error.

Reply:
cc @m4theushw

Reply:
And since `GridOverlayWrapper` is used only in `GridOverlays`, I've moved it to `GridOverlays`